### ISSUE TITLE
Users need to be able to update their information after initial application

### ISSUE THREAD

**Issue by TiCampbell** [Get-Your/GetFoco#75]
Users will need to be able to update their information to renew their application, to adjust their AMI range when the new year's values are updated in the application, or when their personal status changes (such as their household size changing).

This is especially important for address changes!

**Comment by TiCampbell** [comment]
Also need to consider how information changes will flow through the program's system (how will coordinators be notified about information changes as opposed to a new applicant?) (tasks for this are not yet created).

**Comment by TiCampbell** [comment]
Note that address changes will need to go through the same verification as the initial application (check for service area, Connexion availability, any future checks after this comment was written).

**Comment by TiCampbell** [comment]
This should be triggered through a new 'update information' (or similar) button on the dashboard, which would lead to a list of steps the user can choose to update information for.

This should use `what_page()` (and the breadcrumb logic) to allow users to go back to specific pages and update their information.

The one difference between the _application_ logic and the _update info_ logic is that update info should bounce the user back to the 'list of steps' instead of moving to the next steps in the application. This way, they can do one at a time, which should be more intuitive.

**Comment by penoptech** [comment]
Financials
Question about years in residence is only applicable to Eligibitly, so it doesn't need to be displayed for an update
Example: Person changes their AMI from 60% to 30%.
Ensure House hold members are updated after change to initial fincial page (moreInfoNeeded)


@TiCampbell: 
QUESTION: Should updating this information update a user's existing qualification status?
QUESTION: Do we want to allow partial updates for the finanical/dependents information? Right now there's nothing stopping a person from leaving after they've updated their number of dependants and income. What's our philosophy on collecting data: "Something is better than nothing" or "All or nothing".
QUESTION: Is the default value and gross annual household ok? See financials.html with the JS AJAX and triggers. And models.py on line 173

**Comment by penoptech** [comment]
@TiCampbell - I've been doing a bunch of testing related to this feature. There were some wrinkles to this that I didn't initially plan for, but I think I have them ironed out now. When we have our next meeting, let's go over my implementation to make sure it'll work.

[Blank diagram.pdf](https://github.com/Get-Your/GetFoco/files/10835678/Blank.diagram.pdf)

**Comment by penoptech** [comment]
Agreed upon implementation is to keep any ACTIVE or PENDING programs as is. If a user's AMI decreases enough though, their qualification status should be reset, so they're eligible to quick apply to applicable programs.

**Comment by TiCampbell** [comment]
Thanks for this, you helped me look at this in a different way. I think the 'information update' can be summed up as such (using current variable names):

- If `AmiRange_max` increases:
    - No variable changes should be made (there are no new programs to qualify for, and therefore `GenericQualified` already reflects the user's verified income)
- If `AmiRange_max` decreases:
    - Set `GenericQualified = 'PENDING'` to trigger a new income verification
    - Change `'NOTQUALIFIED'` to `'PENDING'` for all programs that are available to the user with the new AMI range (or maybe it should be more generic by having this be "Change **anything `not 'ACTIVE'`** to `'PENDING'`")

**Comment by penoptech** [comment]
@TiCampbell - Thank you for this. An interesting question comes up from this implementation... Essentially we're auto enrolling folks in programs if they have a qualifying AMI after updating their information. Why don't we just always auto enroll people in programs if they have a qualifying AMI?

**Comment by TiCampbell** [comment]
Whoops, you're absolutely right - with our current "make available, but don't auto-apply" strategy (for everything but Grocery Rebate with <=30% AMI), the final bullet should be "Change `'NOTQUALIFIED'` to `''`" (I'll edit the comment as well).

After much debate at the beginning of the program, we decided that just making the programs available would give users the dignity of having a choice in their involvement, rather than us effectively showing a 'savior complex' and deciding for them. That being said, we also decided that the Grocery Tax Rebate is too important for residents in the lowest income range to potentially miss, which is why we have the auto-apply.

As an aside, because I've had to adjust how I was looking at this too: since we don't (and likely won't ever) have connections to the backend business systems for the benefits, we can't actually 'auto-enroll' people ('PENDING' in our system is what kicks off a process for the program coordinators to do the actual enrollment; automatically marking 'ACTIVE' would bypass that whole workflow and end up not enrolling the user); the best we can do is 'auto-apply'.
